src/utils.ts

```diff
@@ -519,7 +520,11 @@ export const _UUID = (function () {
         return ret.toString(16)
     }
 
-    return function () {
+    return function (version?: 'v7') {
+        if (version === 'v7') {
```

Comment:
We've been generating these non-standard UUIDs for as long as we've been generating UUIDs. I'd put money on breaking something if I change them all in one go...

Let's give ourselves a way to migrate away from them rather instead of changing wholesale 🙈

Reply:
Thanks for this PR Paul!

AFAIK, capture generates timestamp-sortable UUIDs, and it's a property that helps CH performance (although I'm not sure about the details, we'll need to ask around).
As I wrote on Slack, UUIDv7 indeed looks like a good alternative to [our homegrown UUIDT](https://github.com/PostHog/posthog/blob/master/posthog/models/utils.py#L23)

Reply:
For context, UUIDTs being roughly timestamp-sortable helps Postgres B-tree indexing performance, but NOT ClickHouse due to [this](https://posthog.slack.com/archives/C019RAX2XBN/p1671706553275349)